beacon-chain/sync/initial-sync/service.go

```diff
@@ -61,7 +62,7 @@ type SyncService interface {
 type InitialSync struct {
 	ctx                          context.Context
 	cancel                       context.CancelFunc
-	p2p                          types.P2P
+	p2p                          shared.P2P
```

Comment:
Should we move p2p back to types since it's only getting used by beacon node?

Reply:
Its actually used by the proposer to subscribe to attestation Broadcasts, so it should stay in the shared package

Reply:
Why not use a feed and pass the attestations from beacon node's rpc to proposer?

The question is should we allow validator to touch p2p?

option 1:
p2p <-> sync service<-> block chain service<->rpc service <-> proposer/attester

option 2:
p2p <-> proposer/attester

Option 2 seems easier, but the down side is it puts more work on the validator package (I think we want to keep it relatively lightweight) and there could be duplicated code between proposer and attester.

Any thoughts? @rauljordan  @rawfalafel  @prestonvanloon

Reply:
The security concerns are valid but I am not sure if its possible for our validators to only communicate via rpc, I would imagine attesters that would be handling a shard would need to use p2p for a lot of things. Also by only using rpc we reduce our validators to a one is to one relationship with other actors which might make it difficult for us to represent complex relationships between the different actors down the line

Reply:
> by only using rpc we reduce our validators to a one is to one relationship with other actors

I'm not sure what you mean by this. 

I'm in favor of option 1. The validator is managing private keys; not allowing it to connect directly to the p2p network is a clean abstraction.

Reply:
I was more referring to the fact that attesters would be validating the state of a shard, so they would be interacting with other actors where rpc might not be the most optimal solution, but in the case of the proposer I will change it to rpc then

Reply:
I see. You are referring phase 1 when shard data (w/o execution) comes in to the picture. Even then to interact and play the proof of custody game, attester still needs DB access and it will have to go through with the beacon node via RPC.  If we grant attester DB access, it feels like attester should just become a service in beacon node. Do you guys think attester broadcasts attestation is the right approach here?

Reply:
Should we change it back to `types.P2P`?

Reply:
I actually think it would make more sense to keep it here. The main p2p package is in shared folder so it would make sense to keep the interface there also, if you feel strongly about it I will put the interface back in types